sql/core/src/test/resources/sql-tests/inputs/grouping_set.sql

```diff
@@ -57,3 +57,6 @@ SELECT k1, k2, avg(v) FROM (VALUES (1,1,1),(2,2,2)) AS t(k1,k2,v) GROUP BY GROUP
 SELECT grouping__id, k1, k2, avg(v) FROM (VALUES (1,1,1),(2,2,2)) AS t(k1,k2,v) GROUP BY GROUPING SETS ((k1),(k1,k2),(k2,k1));
 
 SELECT grouping(k1), k1, k2, avg(v) FROM (VALUES (1,1,1),(2,2,2)) AS t(k1,k2,v) GROUP BY GROUPING SETS ((k1),(k1,k2),(k2,k1));
+
+-- grouping_id function
+SELECT grouping_id(k1, k2), avg(v) from (VALUES (1,1,1),(2,2,2)) AS t(k1,k2,v) GROUP BY k1, k2 GROUPING SETS ((k2, k1), k1);
```

Comment:
Hi, @cloud-fan . This seems to be a breaking change because the output of `grouping__id()` function is changed.Is it safe in the old branch?

Reply:
Also, cc @wangyum since he is the release manager of Apache Spark 3.3.1.

Reply:
hmm, then all bug fixes are breaking changes because they change behaviors...

Reply:
And this query fails in 3.3.0, I don't think making a failed query to run is a breaking change.

Reply:
No, there is no query failure, @cloud-fan . What I mean is that this PR change the return value of `grouping_id()`.

Reply:
This is the same situation like `spark.sql.legacy.integerGroupingId`. I believe we need a migration guide and legacy configuration.

Reply:
For example, we can image a query having `WHERE grouping_id(...) = 1`.

Reply:
@dongjoon-hyun Could you provide your case?

Reply:
We have production customers with the above `WHERE grouping_id(...) = ..` pattern SQLs, @wangyum .

Reply:
This patch fixes bugs, but also affects legitimate queries with Spark's built-in method, `grouping_id`. I'll try to make a legacy config today before RC2 cut, @wangyum .

Reply:
Sorry, I've tagged rc2 three hours ago: https://github.com/apache/spark/releases/tag/v3.3.1-rc2

Reply:
@dongjoon-hyun can you provide a test to show that the result of `grouping_id(...)` changes? My test only shows that a failed `group_id(...)` can run after this fix.

Reply:
My bad. I mislead at that time. Here is the PR for that use case.
- https://github.com/apache/spark/pull/38001